netman/adapters/switches/cisco.py

```diff
@@ -137,10 +139,60 @@ def get_interfaces(self):
         interfaces = []
         for data in split_on_bang(self.ssh.do("show running-config | begin interface")):
             interface = parse_interface(data)
-            if interface:
+            if interface is not None and isinstance(interface, Interface):
                 interfaces.append(interface)
         return interfaces
 
+    def get_bonds(self):
+        bonds = []
+        for bond_summary in self._show_etherchannel_summary():
+            if bond_summary:
+                bonds.append(self._bond_from_summary(bond_summary))
+        return bonds
+
+    def get_bond(self, number):
+        for bond_summary in self._show_etherchannel_summary():
+            if bond_summary and int(bond_summary[0]) == number:
+                return self._bond_from_summary(bond_summary)
+        raise UnknownBond(number)
+
+    def add_bond(self, number):
+        if self._has_bond(number):
+            raise BondAlreadyExist(number)
+        with self.config():
+            result = self.ssh.do('interface port-channel {}'.format(number))
```

Comment:
you could starting unsing the result checker introduced in the dell here

https://github.com/internaphosting/netman/blob/master/netman/adapters/switches/dell.py#L258

just a proposition